src/engine/strat_engine/liminal.rs

```diff
@@ -651,6 +651,25 @@ impl LiminalDevices {
             .collect()
     }
 
+    /// Get a mapping of pool UUIDs from all of the LUKS2 devices that are currently
+    /// locked to their key descriptions in the set of pools that are not yet set up.
+    // Precondition: All devices for a given errored pool have been determined to have
+    // the same key description.
+    pub fn locked_pools(&self) -> HashMap<PoolUuid, KeyDescription> {
```

Comment:
Right here, I think this deviates from the logic of ```locked_pool_uuids()``` and I think it shouldn't, because if any devices in the set are still unlocked we need to consider the thing a "locked pool". Also, if you could make ```locked_pool_uuids``` depend on this method perhaps, since I think it should just represent the keys from the result of this method OR get rid of it altogether and rewrite StratEngine's ```locked_pool_uuids()``` method just to get the keys that would tidy things up a bit.

Reply:
Okay this should be resolved!